generic_test.go

```diff
@@ -137,3 +143,77 @@ func TestWithStartupCommand(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, "/tmp/.testcontainers\n", string(content))
 }
+
+func TestGenericReusableContainerInSubprocess(t *testing.T) {
+	containerIDOnce := sync.Once{}
+	containerID := ""
+
+	wg := sync.WaitGroup{}
+	wg.Add(10)
+	for i := 0; i < 10; i++ {
+		go func() {
+			defer wg.Done()
+
+			// create containers in subprocesses, as "go test ./..." does.
+			output := createReuseContainerInSubprocess(t)
+
+			// check is container reused.
+			re := regexp.MustCompile(fmt.Sprintf("%s(.*)%s",
+				"🚧 Waiting for container id ",
+				regexp.QuoteMeta(fmt.Sprintf(" image: %s", nginxDelayedImage)),
+			))
+			match := re.FindStringSubmatch(output)
+
+			containerIDOnce.Do(func() {
+				containerID = match[1]
+			})
+			require.Equal(t, containerID, match[1])
+		}()
+	}
+
+	wg.Wait()
+}
+
+func createReuseContainerInSubprocess(t *testing.T) string {
+	cmd := exec.Command(os.Args[0], "-test.run=TestHelperContainerStarterProcess")
+	cmd.Env = []string{"GO_WANT_HELPER_PROCESS=1"}
+
+	output, err := cmd.CombinedOutput()
+	require.NoError(t, err, string(output))
+
+	return string(output)
+}
+
+// TestHelperContainerStarterProcess is a helper function
+// to start a container in a subprocess. It's not a real test.
+func TestHelperContainerStarterProcess(t *testing.T) {
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "1" {
+		return
```

Comment:
We can use t.Skip here to add an explanation on why this test method is skipped. Wdyt?

Reply:
Sure. Done